Extensions/render/mirror.lisp

```diff
@@ -71,58 +75,93 @@
 	(call-next-method)))))
 
 (defmethod %fill-image :around ((mirror image-mirror-mixin) 
-				x y width height ink background foreground)
+				x y width height ink background foreground clip-region)
+  (when (image-mirror-image mirror)
+    (with-slots (image-lock) mirror
+      (climi::with-lock-held (image-lock)
+	(call-next-method)))))
+
+(defmethod %fill-paths :around ((mirror image-mirror-mixin) paths transformation region ink background foreground)
   (when (image-mirror-image mirror)
     (with-slots (image-lock) mirror
       (climi::with-lock-held (image-lock)
 	(call-next-method)))))
 
-(defmethod %draw-paths :around ((mirror image-mirror-mixin) paths transformation region ink background foreground)
+(defmethod %stroke-paths :around ((mirror image-mirror-mixin) paths line-style transformation region ink background foreground)
   (when (image-mirror-image mirror)
     (with-slots (image-lock) mirror
       (climi::with-lock-held (image-lock)
 	(call-next-method)))))
 
 (defmethod %draw-image ((mirror image-mirror-mixin)
 			image x y width height
-			dst-dx dst-dy
+			to-x to-y
 			clip-region)
+  (when (or (not (rectanglep clip-region))
+            (not (region-contains-region-p clip-region (make-rectangle* to-x to-y (+ to-x width) (+ to-y height)))))
+    (warn "copy image not correct"))
   (let ((region
-	 (copy-image
-	  (image-mirror-image mirror)
-	  image
-	  :x x :y y :width width :height height
-	  :src-dx dst-dx
-	  :src-dy dst-dy)))
+	 (copy-image image x y width height
+                     (image-mirror-image mirror)
+                     to-x to-y)))
     (%notify-image-updated mirror region)))
 
 (defmethod %fill-image-mask ((mirror image-mirror-mixin)
 			     image-mask x y width height x-dest y-dest clip-region ink background foreground)
-  (let ((*background-design* background)
-	(*foreground-design* foreground))
-    (let ((region
-	   (fill-image
-	    (image-mirror-image mirror)
-	    (make-rgba-design ink)
-	    image-mask
-	    :x x :y y :width width :height height
-	    :mask-dx x-dest :mask-dy y-dest)))
-      (%notify-image-updated mirror region))))
+  #+() (when (or (not (rectanglep clip-region))
+            (not (region-contains-region-p clip-region (make-rectangle* x y (+ x width) (+ y height)))))
```

Comment:
for commenting out block of code, use `#+(or)`. Also - is it commented for a reason, or we could safely remove that code?